### ISSUE TITLE
Can't return a Duplex stream from a remote node

### ISSUE THREAD

**Issue by felipegcampos** [moleculerjs/moleculer#325]
## Expected Behavior

It should return a Duplex stream from a remote node.

## Failure Information

You can find details on how to duplicate the issue in the repo https://github.com/felipegcampos/azure-stream-moleculer-issue

### Context

* Moleculer version: 0.13
* NodeJS version: 9
* Operating System: Linux Mint 64-bit

### Failure Logs
```
>> ERROR: stream.pipe is not a function (NodeID: felipe-xps-15-9550-24145)
TypeError: stream.pipe is not a function
    at Service.get (/home/felipe/Documents/Workspace/moleculer/azure-stream-moleculer-issue/assets.service.js:11:14)
    at <anonymous>
```

### Solution

Issue is [here](https://github.com/moleculerjs/moleculer/blob/05d812226039a8bc5e8684bf4b1201d4852448cf/src/transit.js#L754):
`if (data && typeof data.on === "function" && typeof data.read === "function" && typeof data.pipe === "function") {`

Unless you have reasons to do not use it, we should just check with instanceof.

`if (data && data instanceof Stream)`

or if you want to make sure it's a Readable stream we could also use `data.readable === true`. Duplex streams not always seems to have the `read` function but it still working as Readable stream.

or even:

`if (data && typeof data.on === "function" && typeof data.pipe === "function" && data.readable === true)`

[This implementation](https://github.com/Azure/azure-storage-node/blob/66d24e89eeedf77ab83356c69fe5a39ed03b73b9/lib/common/streams/chunkstream.js) for example inherits from Stream and implements its own Duplex. Although it's Readable and Writable, it would fail for Readable, Writable, Duplex and Transform `instanceof` checks.

Share your thoughts,
Tks

**Comment by icebob** [comment]
I used this solution instead of `instanceof Stream` due to https://stackoverflow.com/a/23885304/129346
But `data.readable === true` seems better. Thanks the PR!

**Comment by felipegcampos** [comment]
Yes. I've checked this thread as well.  There are some ways to try to determine if it's actually a Readable stream. Take a look at this project [here](https://www.npmjs.com/package/is-stream). They use `stream.readable !== false && typeof stream._read === 'function' && typeof stream._readableState === 'object';`. But it also return false when checking the azure ChunkStream as readable. I'm not saying the azure ChunkStream is correct but it works as a Readable stream as well.

**Comment by icebob** [comment]
Released in 0.13.1
